subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/BuildScriptBuilder.java

```diff
@@ -1605,4 +1665,82 @@ public void configureConventionPlugin(@Nullable String comment, BlockStatement p
             plugins.add(new PluginSpec("groovy-gradle-plugin", null, comment));
         }
     }
+
+    public interface InsecureProtocolHandler {
+        void handle(URL url, BuildScriptBuilder.ScriptBlockImpl statements);
+
+        static InsecureProtocolHandler forOption(InsecureProtocolOption insecureProtocolOption, BuildInitDsl dsl) {
+            switch (insecureProtocolOption) {
+                case WARN:
+                    return new BuildScriptBuilder.WarningHandler(dsl);
+                case ALLOW:
+                    return new BuildScriptBuilder.AllowingHandler();
+                case UPGRADE:
+                    return new BuildScriptBuilder.UpgradingHandler();
+                default:
+                    throw new IllegalStateException(String.format("Unknown handler: '%s'.", insecureProtocolOption));
+            }
+        }
+    }
+
+    public static class WarningHandler implements InsecureProtocolHandler {
+        private final BuildInitDsl dsl;
+
+        public WarningHandler(BuildInitDsl dsl) {
+            this.dsl = dsl;
+        }
+
+        @Override
+        public void handle(URL url, BuildScriptBuilder.ScriptBlockImpl statements) {
+            final boolean isGroovy = dsl.getId().equalsIgnoreCase("Groovy");
+            final String allowPrefix = isGroovy ? "a" : "isA";
+            final String fileExtension = isGroovy ? "gradle" : "gradle.kts";
+
+            final Logger logger = LoggerFactory.getLogger(BuildScriptBuilder.class);
+            logger.warn("If you wish to use this repository, you will have to uncomment the line '{}llowInsecureProtocol=true' in the generated build.{}} file.", allowPrefix, fileExtension);
```

Comment:
Hmm, this is clever, I wonder if we can make this a little abstract, so we don't have to do this and we can provide a link to more explanations in the user manual:

```suggestion
            logger.warn("Gradle found an insecure protocol in a repository definition. You will have to opt into allowing insecure protocols in the generated build file. See {}.", documentationRegistry.getDocumentationFor("build_init_plugin", "allow_insecure"));
```

This requires getting `DocumentationRegistry` into `BuildScriptBuilder`, but it looks like you can get it easily from `BuildScriptBuilderFactory` which can get it from `ProjectLayoutSetupRegistryFactory`. I also made up the `allow_insecure` section, we would need to add that to the docs as well.